### DISCUSSION TITLE
Issue with responsive design at grapesjs-blocks-basic

### DISCUSSION THREAD

**Discussion by Aadarsh4u-code** [GrapesJS/grapesjs#4680] [Q&A]
Hi there, 
I'm using grapesjs-blocks-basic as a plugin, however, issue is that while I'm trying to make it responsive in mobile view it column does not stack on each other. I Want it should stack one over another in a mobile view

#Its in the desktop view
<img width="1700" alt="pic 1" src="https://user-images.githubusercontent.com/60578237/197119636-dd4321fc-9476-44ba-abe8-d18b8e260d6e.png">

#Its in mobile view
<img width="1701" alt="pic 2" src="https://user-images.githubusercontent.com/60578237/197119704-f2d8b70b-10b9-4c64-b654-c5fa089a04ce.png">
In mobile view, the green column should be below the red. Also Yellow, pink and blue are below one another 

#Code in plugin options
![pic3](https://user-images.githubusercontent.com/60578237/197119743-804d9e96-e19a-440f-9f2a-0946f005878e.png)

i tried the above option found at GitHub but doesn't work for me

#this is code for the device manager
export const deviceManager = {
  devices: [
    {
      name: 'Desktop',
      width: '',
    },
    {
      name: 'Tablet',
      width: '768px', // this value will be used on canvas width
      widthMedia: '810px', // this value will be used in CSS @media
    },
    {
      name: 'Mobile',
      width: '320px',
      widthMedia: '480px',
    },
  ],
};

**Comment by DhanaDSP1120** [comment]
You can use your own blocks using [this ](https://grapesjs.com/docs/modules/Blocks.html#programmatic-usage)

eg : two cell block content

> content: {
      type: 'table',
      attributes: { class: 'twocolbox-table' },
      components:
      {
        type: 'tbody',
        components:
        {
          type: 'row',
          components: [{
            type: 'cell',
            attributes: { class: 'twocolbox-first-cell' },
            styles: `.twocolbox-first-cell{vertical-align: top;padding:3px;width:50%;height: 100px;}@media (max-width: 480px){.twocolbox-first-cell { box-sizing: border-box;vertical-align: top;padding-top: 25px;padding-right: 10px;padding-bottom: 25px;padding-left: 10px;width: 100%;height: 100px;text-align: center;display: table;}}`,
          }, {
            type: 'cell',
            attributes: { class: 'twocolbox-second-cell' },
            styles: `.twocolbox-second-cell{vertical-align: top;padding:3px;width:50%;height: 100px;}@media (max-width: 480px) {.twocolbox-second-cell {box-sizing: border-box; vertical-align: top;padding-top: 25px;padding-right: 10px;padding-bottom: 25px;padding-left: 10px;width: 100%;height: 100px;display: table;text-align: center;}}`,
          }]
        }
      },
      styles: `.twocolbox-table{width: 100%;display:inline-table;padding:3px 3px;}`,
    }

**Reply by Aadarsh4u-code** [reply]
Thank you @DhanaDSP1120 for your response.However I want to use all the default components of grapesjs-basic-block as it is but I only want to change its responsiveness for column1,column2,column3,labelColumn37 without defining it separately.

**Reply by Aadarsh4u-code** [reply]
It was resolved by refactoring above code in plugin options
plugins: [
      (editor) => gjsBlockBasic(editor, { 
        flexGrid: true
      }),
]
